### DISCUSSION TITLE
Ability to distinguish between a supplied null value and a mising field

### DISCUSSION THREAD

**Discussion by threesquared** [swaggest/rest#207] [Q&A]
Given a Struct like this:

```go
type updateReq struct {
   Val      *string `json:"val" required:"false" nullable:"true" enum:"thing,thing1"`
   OtherVal *string `json:"otherval" required:"false" nullable:"true" pattern:"^[a-zA-Z0-9][a-zA-Z0-9.-]+[a-zA-Z0-9]$"`
}
```

If I send something like `{"otherval":null}` it passes validation as the values are not required and nullable but then my decoded input I just get `{Val:<nil> OtherVal:<nil>}` and it seems there is there is no way to work out if the value was omitted from the request or actually provided but as `null`

I want to be able to make this distinction in an update PATCH request so I know if the user wants to set the value to null or leave it as is.

**Comment by vearutop** [comment]
If I remember correctly (not near my laptop now to check), you need a double pointer (`**string`) for such resolution in encoding/json. Please try.
